From bc4eba0bd83ba1d2291e88e901a439873d540761 Mon Sep 17 00:00:00 2001
From: Kate Montgomery <katemontgomery@google.com>
Date: Thu, 26 Jan 2023 18:31:45 +0000
Subject: [PATCH] Fix bypass BAL via LocationManager.requestFlush

Bug: 235823542
Test: atest LocationProviderManagerTest and manual tests
Change-Id: I2a0fa7b99c3ad5ae839d8018ec70cb5c26e33240
(cherry picked from commit 750af79d5ccb282bb79ef40932858fbae801a48b)
Merged-In: I2a0fa7b99c3ad5ae839d8018ec70cb5c26e33240
---
 .../server/location/provider/LocationProviderManager.java        | 1 +
 1 file changed, 1 insertion(+)

diff --git a/services/core/java/com/android/server/location/provider/LocationProviderManager.java b/services/core/java/com/android/server/location/provider/LocationProviderManager.java
index 8955c288391f..f29c737e60dc 100644
--- a/services/core/java/com/android/server/location/provider/LocationProviderManager.java
+++ b/services/core/java/com/android/server/location/provider/LocationProviderManager.java
@@ -258,6 +258,7 @@ public class LocationProviderManager extends
         public void deliverOnFlushComplete(int requestCode) throws PendingIntent.CanceledException {
             BroadcastOptions options = BroadcastOptions.makeBasic();
             options.setDontSendToRestrictedApps(true);
+            options.setPendingIntentBackgroundActivityLaunchAllowed(false);
 
             mPendingIntent.send(mContext, 0, new Intent().putExtra(KEY_FLUSH_COMPLETE, requestCode),
                     null, null, null, options.toBundle());
-- 
2.39.1.456.gfc5497dd1b-goog

